airflow/contrib/operators/mysql_to_gcs.py

```diff
@@ -160,26 +166,33 @@ def _write_local_schema_file(self, cursor):
             contains the BigQuery schema fields in .json format.
         """
         schema = []
-        for field in cursor.description:
-            # See PEP 249 for details about the description tuple.
-            field_name = field[0]
-            field_type = self.type_map(field[1])
-            # Always allow TIMESTAMP to be nullable. MySQLdb returns None types
-            # for required fields because some MySQL timestamps can't be
-            # represented by Python's datetime (e.g. 0000-00-00 00:00:00).
-            field_mode = 'NULLABLE' if field[6] or field_type == 'TIMESTAMP' else 'REQUIRED'
-            schema.append({
-                'name': field_name,
-                'type': field_type,
-                'mode': field_mode,
-            })
+        tmp_schema_file_handle = NamedTemporaryFile(delete=True)
+        if self.schema is not None and isinstance(self.schema, (str, unicode)):
```

Comment:
unicode is PY2 specific, recommend checking isinstance(self.schema, dict) or use the six library:
```
from six import string_types
isinstance(s, string_types)
```